testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java

```diff
@@ -2994,6 +3011,125 @@ public void whenUpdateDynamicClientDuringRemainingExpirationPeriodMustRotateSecr
 
     }
 
+    @Test
+    public void testIntentClientBindCheck() throws Exception {
+        final String intentName = "openbanking_intent_id";
+
+        // register profiles
+        String json = (new ClientProfilesBuilder()).addProfile(
+                (new ClientProfileBuilder()).createProfile(PROFILE_NAME, "Het Eerste Profiel")
+                        .addExecutor(IntentClientBindCheckExecutorFactory.PROVIDER_ID,
+                                createIntentClientBindCheckExecutorConfig(intentName, TestApplicationResourceUrls.checkIntentClientBoundUri()))
+                        .toRepresentation()
+        ).toString();
+        updateProfiles(json);
+
+        // register policies
+        json = (new ClientPoliciesBuilder()).addPolicy(
+                (new ClientPolicyBuilder()).createPolicy(POLICY_NAME, "Het Eerste Beleid", Boolean.TRUE)
+                        .addCondition(ClientScopesConditionFactory.PROVIDER_ID,
+                                createClientScopesConditionConfig(ClientScopesConditionFactory.OPTIONAL, Arrays.asList("microprofile-jwt")))
+                        .addProfile(PROFILE_NAME)
+                        .toRepresentation()
+        ).toString();
+        updatePolicies(json);
+
+        // create a client
+        String clientId = generateSuffixedName(CLIENT_NAME);
+        String clientSecret = "secret";
+        createClientByAdmin(clientId, (ClientRepresentation clientRep) -> {
+            clientRep.setSecret(clientSecret);
+            clientRep.setStandardFlowEnabled(Boolean.TRUE);
+            clientRep.setImplicitFlowEnabled(Boolean.TRUE);
+        });
+        ClientResource app = findClientResourceByClientId(adminClient.realm("test"), clientId);
+        ProtocolMappersResource res = app.getProtocolMappers();
+        res.createMapper(ModelToRepresentation.toRepresentation(ClaimsParameterWithValueIdTokenMapper.createMapper("claimsParameterWithValueIdTokenMapper", "openbanking_intent_id", true))).close();
+
+        // register a binding of an intent with different client
+        String intentId = "123abc456xyz";
+        String differentClientId = "test-app";
+        Response r = testingClient.testApp().oidcClientEndpoints().bindIntentWithClient(intentId, differentClientId);
+        assertEquals(204, r.getStatus());
+
+        // create a request object with claims
+        String nonce = "naodfejawi37d";
+        Map<String, Object> claim = new HashMap<>();
```

Comment:
As pointed above, it can be useful for re-use the `ClaimsRepresentation` class. Maybe it is useful to add some util method to the `ClaimsRepresentation` class like `public void setClaimValue(String claimName, Object claimValue, ClaimContext ctx)` or something like that?

Reply:
Yes, I will use `ClaimsRepresentation`.